lib/config/configuration.go

```diff
@@ -1135,6 +1135,22 @@ func applyDatabasesConfig(fc *FileConfig, cfg *service.Config) error {
 				Tags:    matcher.Tags,
 			})
 	}
+	for _, matcher := range fc.Databases.AzureMatchers {
+		if len(matcher.Subscriptions) == 0 || apiutils.SliceContainsStr(matcher.Subscriptions, types.Wildcard) {
+			matcher.Subscriptions = []string{types.Wildcard}
+		}
+		if len(matcher.ResourceGroups) == 0 || apiutils.SliceContainsStr(matcher.ResourceGroups, types.Wildcard) {
+			matcher.ResourceGroups = []string{types.Wildcard}
+		}
+		cfg.Databases.AzureMatchers = append(cfg.Databases.AzureMatchers,
```

Comment:
This fixup is a bit strange; I'd expect this logic of equivalence between `len(matcher.Subscriptions) == 0` and `matcher.Subscriptions = []string{types.Wildcard}` to be part of core matching logic, but that is not part of this PR. Other matchers, AWS in particular, do not have this sort of logic applied here, which I think is a good choice.

Reply:
The **Contains** check seems an optimization. I don't think we have done it for AWS Matchers (for ex: deduplicate regions). But I also agree that this logic belongs where matching happens.

Reply:
The larger context PR was enormous so I split it up.

I think this fixup is about configuration, not about matching logic. The contains check is indeed an optimization, and I should also include deduplication, but I think if two configurations are equivalent it makes sense to simplify the config as soon as possible. It's also much simpler to write and read tests for configuration equivalence than for matching behavior equivalence. I don't feel too strongly either way though, so I'm open to moving it if you still disagree.

Reply:
My concern is this code, which defines the semantics of empty resource group match, is not guaranteed to be applied consistently in all cases or be kept in the future.

For example, I expect the `AzureMatcher` to be used in the future for matching VMs and not just dbs. The person implementing that would have to copy these few lines. If they fail to do so, the behavior of matchers will diverge. 

In general, it is hard to ensure how the `AzureMatcher` struct will be constructed in the future; it makes sense to ensure that it behaves in the same way independent of the way that happened.

Reply:
Ok I'll move it to the code using the matchers. I agree that locality is good - keep code that affects other code closer together. However, this logic will still need to be duplicated for azure VMs or other resources.